kotest-assertions/src/jvmMain/kotlin/io/kotest/matchers/equality/reflection.kt

```diff
@@ -190,14 +184,14 @@ fun <T : Any> beEqualToIgnoringFields(
    override fun test(value: T): MatcherResult {
 
       val fieldNames = fields.map { it.name }
-      val failed = value::class.memberProperties.filterNot { fieldNames.contains(it.name) }.mapNotNull {
-         val actual = it.getter.call(value)
-         val expected = it.getter.call(other)
-         if (actual == expected) null else {
-            "${it.name}: $actual != $expected"
+      val fieldsToBeConsider: List<KProperty<*>> = value::class.memberProperties
+         .filterNot { fieldNames.contains(it.name) }
+         .map {
+            it.isAccessible = true
+            it
```

Comment:
In the end, this just maps to `it` (i.e. identity) as the line `it.isAccessible = true` is ignored here. What you probably want is:

```
val fieldsToBeConsiderd = value::class.memberProperties.filter {
    !fieldNames.contains(it.name) && it.isAccessible
}
```

Reply:
No, ```it.isAccessible = true``` is not getting ignored, it is making ever other field accessible(not considering if they are already accessible) which are not asked to ignore which checking equality

Reply:
Ah, sorry, I was misreading the assignment. So how about using `onEach` instead of a "dummy" `map` then?

Reply:
done

Reply:
You're using `forEach` now, which returns `Unit`. With `onEach` you could have kept the fluent API as it returns the modified list.

Reply:
OK I was not aware of ```onEach```